v2/pkg/protocols/headless/engine/rules.go

```diff
@@ -51,4 +52,32 @@ func (p *Page) routingRuleHandler(ctx *rod.Hijack) {
 			ctx.Response.SetBody(rule.Args["body"])
 		}
 	}
+
+	// store history
+	req := ctx.Request.Req()
+	var rawReq string
+	if raw, err := httputil.DumpRequestOut(req, true); err == nil {
+		rawReq = string(raw)
+	}
+
+	// attempts to rebuild the response
+	var rawResp strings.Builder
+	respPayloads := ctx.Response.Payload()
+	if respPayloads != nil {
+		rawResp.WriteString("HTTP/1.1 ")
+		rawResp.WriteString(fmt.Sprint(respPayloads.ResponseCode))
+		rawResp.WriteString(" " + respPayloads.ResponsePhrase + "+\n")
```

Comment:
@Mzack9999 

I think in the `"+\n"`, the `+` sign is a mistake. 

Also his might be more readable:

```go
rawResp.WriteString(fmt.Sprintf("HTTP/1.1 %d %s\n", respPayloads.ResponseCode, respPayloads.ResponsePhrase))
```

Reply:
> @Mzack9999
> 
> I think in the `"+\n"`, the `+` sign is a mistake.
> 

Nice catch - fixed in https://github.com/projectdiscovery/nuclei/pull/1465

> Also his might be more readable:
> 
> ```go
> rawResp.WriteString(fmt.Sprintf("HTTP/1.1 %d %s\n", respPayloads.ResponseCode, respPayloads.ResponsePhrase))
> ```

^ @Ice3man543 - This was implemented as requested in https://github.com/projectdiscovery/nuclei/pull/1432#discussion_r776676787

Reply:
I must disagree with that comment. In the original version (which actually coincides with my recommendation ;-) ), it's clearly visible that it describes one HTTP request line. Splitting it into multiple lines, not only makes the code longer (without any relevant performance benefit), but clutters the logic and makes it harder to follow.